Add manual testing (#239)
AY2122S1-CS2103T-W16-2/tp: docs/DeveloperGuide.md

</div>

### Recipe

1. Deleting a recipe ingredient
    1. Prerequisites: Recipe at index 1 has at least 1 ingredient and at most 3 ingredients.
    1. Test case: `delete-ri 1 i/1`
       <br>Expected: First ingredient is deleted from the ingredients in the first recipe of the displayed recipe bookmarks list.
       Details of the deleted ingredient and edited recipe is shown in the result display box.
    1. Test case: `delete-ri 1 i/10`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri abc i/1`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri 1 i/abc`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri 1`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri i/1`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
1. Editing a recipe ingredient

>     1. Prerequisites: Recipe bookmarks list shows at least 1 recipe and at most 3 recipes.

Reviewer: Need to include that there's no existing duplicate recipe? And test case failure when trying to edit a recipe when it already exists
```suggestion
    1. Prerequisites: Recipe bookmarks list only has these 3 recipes:
        * name: Pecan pie, price: $10
        * name: Banana split, price: $10
        * name: Orange cake, price: $12
```

Author: Oh that's true. I think it's better to not include those cases because it becomes additional work to explain.

Reviewer: Ok that's fine. Anyways this is just a guide, it doesn't need to be exhaustive.

    1. Test case: `edit-r 1 rn/Truffle fries rp/8.00`
       <br>Expected: First recipe in the recipe bookmarks list is edited to have recipe name 'Truffle fries' and recipe price of '$8.00'.

>        Its position in the recipe bookmarks list may change. Details of the edited recipe shown in the result display box.

Reviewer 2: ```suggestion
       Its position in the recipe bookmarks list may change. Details of the edited recipe is shown in the result display box.
```

    1. Test case: `edit-r 1 rn/Apple pie`
       <br>Expected: First recipe in the recipe bookmarks list is edited to have recipe name 'Apple pie'.

>        Its position in the recipe bookmarks list may change. Details of the edited recipe shown in the result display box.

Reviewer 2: ```suggestion
       Its position in the recipe bookmarks list may change. Details of the edited recipe is shown in the result display box.
```

    1. Test case: `edit-r rn/Apple pie`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `edit-r`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `edit-r 1`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `edit-r abc`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `edit-r 1 rp/-2`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
1. Finding recipe by keywords
    1. Prerequisites: There are exactly 2 recipes in the recipe bookmarks list.
       One has the recipe name "Apple Pie", the other has the recipe name "Banana split".
    1. Test case: `find-r rn/apple`

>        <br>Expected: Recipe bookmarks list only shows the recipe with the recipe name "Apple Pie".

Reviewer 2: ```suggestion
       <br>Expected: Recipe bookmarks list only shows the recipe with the recipe name 'Apple Pie'.
```

       <br>Expected: First ingredient is deleted from the ingredient list in the first recipe of the displayed recipe bookmarks list.
       Details of the deleted ingredient and edited recipe are shown in the result display box.
    1. Test case: `delete-ri 1 i/10`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri abc i/1`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri 1 i/abc`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri 1`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri i/1`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
    1. Test case: `delete-ri`
       <br>Expected: No recipe is edited. Error details shown in the result display box.
1. Editing a recipe
    1. Prerequisites: Recipe bookmarks list only has these 3 recipes:
        * name: Banana split, price: $10
        * name: Orange cake, price: $12
        * name: Pecan pie, price: $10

>      1. Test case: `edit-r 1 rn/Truffle fries rp/8.00`

Reviewer: The '1.' is one space behind
